### PR TITLE
CI: publish benchmarks log as artifact

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
To check the benchmarks timings for https://github.com/pandas-dev/pandas/issues/44450

### CODE REVIEW

**Comment by jreback** [comment]
where does this show up?

**Comment by jorisvandenbossche** [comment]
Well, it doesn't yet show up anywhere, because with the updated asv comment it complains about having no machine information (see failing ci of latest commit), and not directly sure how to fix that.

**Comment by jorisvandenbossche** [comment]
Potentially related upstream issue: https://github.com/airspeed-velocity/asv/issues/944

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -84,7 +84,7 @@ jobs:
         git remote add upstream https://github.com/pandas-dev/pandas.git
         git fetch upstream
         asv machine --yes
-        asv dev | sed "/failed$/ s/^/##[error]/" | tee benchmarks.log
+        asv run --quick --show-stderr --python=same --set-commit-hash HEAD | sed "/failed$/ s/^/##[error]/" | tee benchmarks.log
```
Comment (lithomas1) [inline .github/workflows/ci.yml:87]: This change should be unnecessary. I think the asv docs say that ``asv dev`` is
equivalent to ``asv run --quick --show-stderr --python=same``

Comment (jorisvandenbossche) [reply .github/workflows/ci.yml:87]: Yes, but in practice it's not exactly like that. Here specifically I needed to specify `--set-commit-hash` to have it save results (that is documented), and this keyword is not passed through by `asv dev`.

Comment (lithomas1) [reply .github/workflows/ci.yml:87]: Thanks for the clarification.

**Comment by jorisvandenbossche** [comment]
So the problem with using `asv run` instead of `asv dev` is that it now gives *"No information stored about machine 'fv-az186-290'. I know about nothing"*. Not directly sure how to work-around that (see also https://github.com/airspeed-velocity/asv/issues/944)

**Comment by jreback** [comment]
still active?

**Comment by mroeschke** [comment]
Looks like this currently isn't possible due to airspeed-velocity/asv#944. Once the upstream issue is address then this approach can be continued. Closing for now.

**Pull request closed without merging**